File: backend-v3/models/acquirerConfig.js
'use strict'

module.exports = (sequelize, DataTypes) => {
  const acquirerConfig = sequelize.define('acquirerConfig', {
    name: DataTypes.STRING,
    description: DataTypes.STRING,

    handler: DataTypes.STRING,

    sandbox: DataTypes.BOOLEAN,
    merchantId: DataTypes.INTEGER,

    config: DataTypes.JSONB
  }, {
    timestamps: true,
    freezeTableName: true,
    deletedAt: 'archivedAt',
    paranoid: true
  })

  acquirerConfig.associate = (models) => {
    acquirerConfig.belongsTo(models.merchant, { foreignKey: 'merchantId' })
    acquirerConfig.hasMany(models.acquirerConfigAgent, { foreignKey: 'acquirerConfigId' })
    acquirerConfig.hasMany(models.acquirerConfigOutlet, { foreignKey: 'acquirerConfigId' })
  }

  acquirerConfig.addScope('excludeConfig', {
    attributes: {
      exclude: [
        'config'
      ]
    }
  })
  acquirerConfig.addScope('admin', () => ({
    paranoid: false
  }))

  return acquirerConfig
}

File: frontend-mobile-v3/sdk/src/main/java/id/mikaapp/sdk/models/AcquirerConfig.kt
package id.mikaapp.sdk.models

import com.google.gson.annotations.SerializedName

data class AcquirerConfig(
    @SerializedName("description")
    val description: Any,
    @SerializedName("handler")
    val handler: String,
    @SerializedName("id")
    val id: String,
    @SerializedName("merchantId")
    val merchantId: String,
    @SerializedName("name")
    val name: String,
    @SerializedName("sandbox")
    val sandbox: Boolean
)